### PR TITLE
organizations: take into account `companyd` orgs when creating/updating/deleting org CRs

### PR DESCRIPTION
**Pull request by axbarsan**
Closes https://github.com/giantswarm/giantswarm/issues/17121

This PR adds support for ensuring `companyd` organizations whenever `Organization` CRs get created or updated. It also will delete `companyd` organizations when `Organization` CRs get deleted, but only if there are no credentials set up.

### CODE REVIEW

File: service/controller/key/key.go [hunk 1]
```diff
@@ -19,3 +20,14 @@ func ToOrganization(v interface{}) (securityv1alpha1.Organization, error) {
 
 	return *c, nil
 }
+
+func LegacyOrganizationName(cr *securityv1alpha1.Organization) string {
+	if cr.GetAnnotations() != nil {
+		name := cr.GetAnnotations()[annotation.UIOriginalOrganizationName]
```
Comment (axbarsan) [inline service/controller/key/key.go:26]: The annotation is used for `companyd` orgs that have names which don't adhere to the DNS spec. The annotation holds the `companyd` org name

File: service/controller/resource/organization/delete.go [hunk 2]
```diff
@@ -16,6 +19,40 @@ func (r *Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
...
+		legacyCredentialRequest := legacyCredentialLister.Request{
+			Organization: legacyOrgName,
+		}
+		legacyCredentials, err = r.legacyCredentialClient.List(ctx, legacyCredentialRequest)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	if len(legacyCredentials) > 0 {
```
Comment (axbarsan) [inline service/controller/resource/organization/delete.go:37]: Prevent deleting orgs that have legacy credentials. If we don't, someone else could create another organization with the same name and use the same credentials

**Review by corest — APPROVED** [review]
I don't like having this in organization-operator. But if that's the only way - fine.
Do you also remove legacy organization creation from API then?

**Comment by axbarsan** [comment]
> Do you also remove legacy organization creation from API then?

Not at this time, as it doesn't collide with the implementation in the operator

File: service/controller/resource/organization/delete.go (hunk 2 above)
Comment (kopiczko) [reply service/controller/resource/organization/delete.go:37]: Please put that as a comment in the code, because it isn't easy to realize.

**Pull request merged**